### ISSUE TITLE
orderByUsingJoins fails when called in a 2nd query for the same model

### ISSUE THREAD

**Issue by georgeboot** [kirschbaum-development/eloquent-power-joins#16]
We have the following code:
```php
$user = auth()->user();

$query1 = $user->comments()
    ->orderByUsingJoins('post.created_at', 'desc')
    ->get();

$query2 = $user->comments()
    ->orderByUsingJoins('post.created_at', 'desc')
    ->get();
```

Query 1 runs successfully. However, query 2 fails:
```
Unknown column 'posts.created_at' in 'order clause'
```
This happens because the `join` statement is missing from the 2nd query, while the `order by` part is still present.

Calling `$user->refresh()` between the two queries, works around this issue.

**Comment by luisdalmolin** [comment]
@georgeboot Interesting, I would not expect this to happen. Definitely going to take a look at this, probably over the weekend.

**Comment by luisdalmolin** [comment]
@georgeboot It took longer than I would like but had some crazy last weeks. This should now be fixed in version 2.0. Make sure you check the new method signature for 2.0. as well that you now have to use a trait in your model. This version is already ready to support Laravel 8 as well.
